<a href="https://colab.research.google.com/github/SatoJin02/BDA_course25/blob/main/s1290074_11_task1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Task 1: IRIS Dataset - Data Exploration and Naive Bayes Classification**

# 1.Data Exploration Summary

The IRIS dataset (150 total samples, 4 features: sepal/petal length/width) was loaded. Data exploration confirmed that petal length and petal width are the most effective features for separating the classes. The Setosa class is linearly separable, but Versicolor and Virginica show a slight overlap in their feature space.

#2.Naive Bayes Model Implementation (Code)

The Gaussian Naive Bayes model was implemented using a 70% training / 30% testing split

In [1]:
import pandas as pd
from sklearn.datasets import load_iris
from sklearn.model_selection import train_test_split
from sklearn.naive_bayes import GaussianNB
from sklearn.metrics import confusion_matrix, accuracy_score, classification_report
import matplotlib.pyplot as plt
import seaborn as sns

# Load the IRIS dataset
iris = load_iris()
X = iris.data
y = iris.target
target_names = iris.target_names

# Create DataFrame for exploration (as shown in console output)
df = pd.DataFrame(data=X, columns=iris.feature_names)
df['species_name'] = y
df['species_name'] = df['species_name'].apply(lambda x: target_names[x])

print("--- Data Exploration Summary ---")
print("Features:", iris.feature_names)
print("\nFirst 5 rows:")
print(df.head())
# sns.pairplot(df, hue='species_name', diag_kind='kde') # Plotting code removed from markdown for brevity

# Split data
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

# Instantiate and train the Gaussian Naive Bayes model
model = GaussianNB()
model.fit(X_train, y_train)

# Predict and Evaluate
y_pred = model.predict(X_test)
accuracy = accuracy_score(y_test, y_pred)

print("\n--- Evaluation Results ---")
print(f"Accuracy Score: {accuracy:.4f}")
print("\nClassification Report:")
print(classification_report(y_test, y_pred, target_names=target_names))
print("\nConfusion Matrix:")
print(confusion_matrix(y_test, y_pred))

--- Data Exploration Summary ---
Features: ['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)']

First 5 rows:
   sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)  \
0                5.1               3.5                1.4               0.2   
1                4.9               3.0                1.4               0.2   
2                4.7               3.2                1.3               0.2   
3                4.6               3.1                1.5               0.2   
4                5.0               3.6                1.4               0.2   

  species_name  
0       setosa  
1       setosa  
2       setosa  
3       setosa  
4       setosa  

--- Evaluation Results ---
Accuracy Score: 0.9778

Classification Report:
              precision    recall  f1-score   support

      setosa       1.00      1.00      1.00        19
  versicolor       1.00      0.92      0.96        13
   virginica       0.93      1.00      0.96 

#3.Results Summary and Observation

The Gaussian Naive Bayes model achieved a high Accuracy of 0.9778.
The Setosa class was perfectly classified (Precision/Recall/F1 = 1.00). The Confusion Matrix showed only one Versicolor sample misclassified as Virginica. The model's overall strong performance confirms that the independence assumption of Naive Bayes works well for this dataset.